packages/auth-plugins/src/AuthPluginPsk.tsx

```diff
@@ -1,66 +1,216 @@
-import React, { useEffect, useState } from 'react';
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import { CSSTransition } from 'react-transition-group';
+import { LoadingOverlay, ThemeExport } from '@deephaven/components';
+import { useClient } from '@deephaven/jsapi-bootstrap';
+import { useBroadcastLoginListener } from '@deephaven/jsapi-components';
 import Log from '@deephaven/log';
-import { LoadingOverlay } from '@deephaven/components';
+import { getErrorMessage } from '@deephaven/utils';
+import Cookies from 'js-cookie';
 import { AuthPlugin, AuthPluginProps } from './AuthPlugin';
+import LoginForm from './LoginForm';
+import Login from './Login';
+import AuthenticationError from './AuthenticationError';
+
+const AUTH_TYPE = 'io.deephaven.authentication.psk.PskAuthenticationHandler';
+
+const PSK_QUERY_PARAM_KEY = 'psk';
+
+const PSK_TOKEN_KEY = 'io.deephaven.web.client.auth.psk.token';
 
 const log = Log.module('AuthPluginPsk');
 
-const AUTH_TYPE = 'io.deephaven.authentication.psk.PskAuthenticationHandler';
+function getWindowToken(): string | null {
+  return new URLSearchParams(window.location.search).get(PSK_QUERY_PARAM_KEY);
+}
+
+function clearWindowToken() {
+  log.debug2('clearWindowToken');
+  const params = new URLSearchParams(window.location.search);
+  params.delete(PSK_QUERY_PARAM_KEY);
+
+  let queryString = `${params}`;
+  if (queryString.length > 0) {
+    queryString = `?${queryString}`;
+  }
+  const newPath = `${window.location.pathname}${queryString}${window.location.hash}`;
+  window.history.replaceState(null, '', newPath);
```

Comment:
```suggestion
  const url = new URL(window.location);
  url.searchParams.delete(PSK_QUERY_PARAM_KEY);

  window.history.replaceState(null, '', url.toString());
```

Reply:
Geez I really took the stupid way with this one, good cleanup.